### PR TITLE
Cloning simulations

### PR DESCRIPTION
**Pull request by abdelr**
WIP #521 (Clone simulations)

### CODE REVIEW

**Comment by abdelr** [comment]
@PavelBal @msevestre and @Yuri05 I have marked this as a draft because I am not sure about the whole specifications. Currently, it does clone the simulation based on the entity to clone but maybe we would like to make a deeper or shallower copy. I am struggling a bit here since we have not worked too much into MoBi direction yet so please, take a look and let me know what do you think is still missing or completely wrong. Also, the tests are for sure, not complete. Here I would like to hear a bit more about what are the expectations after cloning a simulation so I could turn those requirements into tests ;).

**Comment by msevestre** [comment]
Let's postpone this until after the release. Cloning is far from trivial because of all the references to building blocks going on. The implementation might be working but it would require a lot of testing. I'll look at this later

**Comment by PavelBal** [comment]
Yes, definitely not for V10. We'll continue working on it in the branch, it is also a good opportunity to learn more about how simulations are set up in MoBi.

**Comment by abdelr** [comment]
Definitely not for V10. I think we all agreed V10 should be sealed now (unless some bug fixing). I also have the feeling this is going to take some time and testing and this is why I opened the discussion ;)

File: src/MoBi.Presentation/Tasks/Edit/EditTaskFor.cs [hunk 1]
```diff
@@ -42,22 +50,59 @@ public abstract class EditTaskFor<T> : IEditTaskFor<T> where T : class, IObjectB
...
+         _objectBaseFactory = objectBaseFactory;
+         _cloneManagerForModel = cloneManagerForModel;
       }
 
       public virtual void Edit(T objectToEdit)
       {
          _context.PublishEvent(new EntitySelectedEvent(objectToEdit, this));
       }
 
+      protected TEntity Clone<TEntity>(TEntity entityToClone) where TEntity : class
```
Comment (msevestre) [inline src/MoBi.Presentation/Tasks/Edit/EditTaskFor.cs:85]: @abdelr This is not correct. A clone method like this one will keep all internal ids resulting in different object with same id. I know this is a draft but you cannot rely on this method

Comment (msevestre) [reply src/MoBi.Presentation/Tasks/Edit/EditTaskFor.cs:85]: I open the PR by mistake. Let me know when you want me to review

**Comment by georgeDaskalakis** [comment]
@msevestre is this PR up to date? should we actually delet or is there something to finalize?

**Comment by msevestre** [comment]
I know nothing about this PR. Maybe @PavelBal knows what the status is?

**Comment by PavelBal** [comment]
Abdel started to work on implementation of simulation cloning in MoBi, but it was not finished. Cloning simulations is an important feature, however, it might get obsolete with OSMOSES. If it does not bother you, we could let this PR live for a while until we either a) solve this in OSMOSES or b) decide to implement it separately.

**Comment by PavelBal** [comment]
Let's close the PR, but keep the branch for now.

**Pull request closed without merging**